### PR TITLE
Created AI + Single player

### PR DESCRIPTION
**Pull request by Miku1338**
This is a working version of the singleplayer, although very buggy and doesn't seem to show the updates to the enemies from the AI's functions.  But at least it runs and is more interesting than just aimlessly walking around in singleplayer.  Sorry it's not perfect yet but I think it's progress.

### CODE REVIEW

File: game/src/js/objects/Character.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ export default class Character extends Phaser.Physics.Arcade.Sprite {
       this.setVelocity(vector.x * this.props.baseSpeed, vector.y * this.props.baseSpeed);
     }
     let animation = 'stance';
-    if(vector.length() != 0) {
+    if(vector.length != 0) {
```
Comment (ignacioxd) [inline game/src/js/objects/Character.js]: I think length is not a property of vector, but instead a method as it was originally. Could you please change this back?

File: game/src/js/objects/FireMonster.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import Character from './Character';
+
+export default class FireMonster extends Character {
```
Comment (ignacioxd) [inline game/src/js/objects/FireMonster.js]: We removed these Character subclasses and are now using the `Character` class directly configured from JSON.  Please use the `Character` class directly.

File: game/src/js/scenes/DungeonScene.js [hunk 3]
```diff
@@ -67,10 +68,28 @@ export default class DungeonScene extends BaseScene {
...
   }
 
   update() {
+    --this.delay;
+    ++this.virtualTime;
+    if (this.delay <= 0) {
+      var newMonster = new FireMonster(this, 250 * (Math.random() - 0.5), 250 * (Math.random() - 0.5));
+      newMonster.setAI(this.localCharacter);
+      this.enemyList.push(newMonster);
+      this.characters.add(newMonster);
```
Comment (ignacioxd) [inline game/src/js/scenes/DungeonScene.js]: Looks like this is adding the character to this list properly, but no action is taken when a collision occurs. Could you add a collision handler? See line 24.

File: game/src/js/scenes/DungeonScene.js [hunk 4]
```diff
@@ -67,10 +68,28 @@ export default class DungeonScene extends BaseScene {
...
+    this.delay = 50;
+    this.enemyList = [];
+    this.virtualTime = 0;
   }
 
   update() {
+    --this.delay;
+    ++this.virtualTime;
+    if (this.delay <= 0) {
+      var newMonster = new FireMonster(this, 250 * (Math.random() - 0.5), 250 * (Math.random() - 0.5));
```
Comment (ignacioxd) [inline game/src/js/scenes/DungeonScene.js]: This should be instantiating the `Character` class, perhaps with either a random character type of having probabilities for spawning each character type.

File: game/src/js/objects/Character.js (hunk 1 above)
Comment (Miku1338) [reply game/src/js/objects/Character.js]: It crashes the game the way it current was.  I don't think I can get it to work without changing it.

File: game/src/js/objects/FireMonster.js (hunk 2 above)
Comment (Miku1338) [reply game/src/js/objects/FireMonster.js]: I know that, I just couldn't figure out how to make it work that way.  I can try again though.

File: game/src/js/scenes/DungeonScene.js (hunk 4 above)
Comment (Miku1338) [reply game/src/js/scenes/DungeonScene.js]: Yes I know that's ideal, I just didn't really see how to do that.

File: game/src/js/objects/Character.js (hunk 1 above)
Comment (ignacioxd) [reply game/src/js/objects/Character.js]: According to the docs, this should be a method: https://photonstorm.github.io/phaser3-docs/Phaser.Math.Vector2.html#length.

File: game/src/js/objects/Character.js [hunk 5]
```diff
@@ -200,4 +200,54 @@ export default class Character extends Phaser.Physics.Arcade.Sprite {
...
+    } else if (xDifference < 0) {
+      xChange = 10;
+    }
+    if (yDifference > 0) {
+      yChange = -10;
+    } else if (yDifference < 0) {
+      yChange = 10;
+    }
+    const vector = [xChange, yChange];
+    this.setMotion(vector);
```
Comment (ignacioxd) [inline game/src/js/objects/Character.js]: Here's the problem with the vector. This is passing in an array instead of a Phaser.Math.Vector2

**Pull request merged**